### PR TITLE
Temporary fix for #301

### PR DESCRIPTION
**Pull request by niconoe-**
Like I said in #301 this is only a temporary fix to avoid the deprecation error. The real work to do to prepare PHP 9.0 correctly is to infer the good return type, which is not OK with interfaces currently.

### CODE REVIEW

**Review by 20uf — APPROVED** [review]

**Comment by adoy** [comment]
Hi,

Thanks for the pull request. Do you know if there is any impact by adding this Annotation on every method ?
I'll have a deeper look to see if this may cause problems in the future if we add this annotation temporarily.

**Comment by niconoe-** [comment]
I tried it by modifying my instance of your library directly on my project and got no bad surprised. 

I remarked that if you put this attribute to a method that doesn't need it, there's no notice/warning/errors, so I feel it very safe.

The only negative point to me is that it masks the deprecation so, when PHP 9 will be on tracks, there will be no way to remember this is a temporary fix. But I hope someone more experienced than me with Phake could find a way to be aligned with the PHP RFCs until then 😃

**Comment by adoy** [comment]
I think it would be best if we only add the `#[\ReturnTypeWillChange]` attribute only if the mocked method is *internal*.

What do you think ?

```
if (PHP_VERSION_ID >= 80100 && $method->isInternal())
```

Also since we already know that this deprecation is only for 8.1+ to 9.0 what about adding a check to make sure `PHP_VERSION_ID is <= 90000` ?

**Comment by niconoe-** [comment]
Yeah, I think this is something that could work. I would like to give it a try on my project at work, especially for the `$method->isInternal()`, as for PHP 9 condition, I agree with you, it's better to limit.

**Comment by adoy** [comment]
👍

**Comment by niconoe-** [comment]
After some tests, nothing went wrong with your proposal. I'll edit the PR.

Thanks!

File: src/Phake/ClassGenerator/MockClass.php [hunk 1]
```diff
@@ -399,10 +400,15 @@ protected function implementMethod(\ReflectionMethod $method, $static = false)
                 $resultReturn = '';
             }
         }
+	elseif (PHP_VERSION_ID >= 80100)
```
Comment (niconoe-) [inline src/Phake/ClassGenerator/MockClass.php:403]:
```suggestion
	elseif (PHP_VERSION_ID >= 80100 && PHP_VERSION_ID < 90000 && $method->isInternal())
```

**Comment by adoy** [comment]
Thanks. Could you please modify your Pull Request so that your change is only one commit?

**Comment by niconoe-** [comment]
I fixed up my modifications to merge the 2 commits into a single one, but I also fetched your master from upstream to be on an up-to-date version of `master`. This causes another commit: 1 from the merge, and 1 from my PR.

Is this OK for you?

File: src/Phake/ClassGenerator/MockClass.php [hunk 2]
```diff
@@ -385,6 +385,7 @@ protected function implementMethod(\ReflectionMethod $method, $static = false)
         }
 
         $returnHint = '';
+	$attributes = '';
```
Comment (adoy) [inline src/Phake/ClassGenerator/MockClass.php:388]: Please use *4 spaces* here as indentation.

File: src/Phake/ClassGenerator/MockClass.php [hunk 3]
```diff
@@ -399,10 +400,15 @@ protected function implementMethod(\ReflectionMethod $method, $static = false)
                 $resultReturn = '';
             }
         }
+	elseif (PHP_VERSION_ID >= 80100 && PHP_VERSION_ID < 90000 && $method->isInternal())
```
Comment (adoy) [inline src/Phake/ClassGenerator/MockClass.php:403]: Same indentation here :-)

File: src/Phake/ClassGenerator/MockClass.php [hunk 4]
```diff
@@ -399,10 +400,15 @@ protected function implementMethod(\ReflectionMethod $method, $static = false)
...
+	{
+	    $attributes = '#[\ReturnTypeWillChange]';
+	}
 
         $docComment = $method->getDocComment() ?: '';
         $methodDef = "
 	{$docComment}
+	{$attributes}
```
Comment (adoy) [inline src/Phake/ClassGenerator/MockClass.php:411]: This indentation is OK ! The generated code uses *tab* for indentation.

**Review by adoy — CHANGES_REQUESTED** [review]
Lasts comments before merging just to fix coding styles.

File: src/Phake/ClassGenerator/MockClass.php (hunk 2 above)
Comment (niconoe-) [reply src/Phake/ClassGenerator/MockClass.php]: I used the GitHub editor to do this and I thought it put 4 spaces... Fixed by using my IDE instead :smile:

File: src/Phake/ClassGenerator/MockClass.php (hunk 3 above)
Comment (niconoe-) [reply src/Phake/ClassGenerator/MockClass.php]: Same issue, same resolving!

**Review by niconoe- — COMMENTED** [review]
Fix bad indentation issues.

**Comment by adoy** [comment]
Thanks for your contribution !

**Pull request merged**